src/explore-education-statistics-admin/src/pages/admin-dashboard/components/__tests__/MethodologySummary.test.tsx

```diff
@@ -424,14 +434,90 @@ describe('MethodologySummary', () => {
     );
   });
 
+  describe('Removing a non-amendment methodology', () => {
+    test('the remove methodology button is shown if user has permission', () => {
+      render(
+        <MemoryRouter>
+          <MethodologySummary
+            publication={testPublicationWithMethodologyCanRemove}
+            topicId={testTopicId}
+            onChangePublication={noop}
+          />
+        </MemoryRouter>,
+      );
+
+      expect(
+        screen.queryByText('Remove', { selector: 'button' }),
+      ).toBeInTheDocument();
+
+      expect(
+        screen.queryByText('Amend methodology', { selector: 'button' }),
+      ).not.toBeInTheDocument();
+    });
+
+    test('shows the confirm modal when clicking the Remove button', async () => {
+      render(
+        <MemoryRouter>
+          <MethodologySummary
+            publication={testPublicationWithMethodologyCanRemove}
+            topicId={testTopicId}
+            onChangePublication={noop}
+          />
+        </MemoryRouter>,
+      );
+
+      userEvent.click(screen.getByText('Remove', { selector: 'button' }));
+      await waitFor(() => {
+        expect(
+          screen.queryByText('Confirm you want to remove this methodology'),
```

Comment:
For things that actually exist on the page, would suggest using `getBy*` selectors rather than the equivalent `queryBy*`

This actually advised by the guy that actually wrote React Testing Library: https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#using-query-variants-for-anything-except-checking-for-non-existence

Reply:
Cool no worries - I figured that if it was within a waitFor() then it wouldn't really matter, as the `expect(...).toBeInTheDocument()` line would fail either way, one for if the `query` variant returned null to `expect` and one for if the `get` call failed because it blows up upon not finding anything.  I guess that the `get` variant's error message is more descriptive potentially?

Begs the question though, why write `expect(getByBlah(....)).toBeInTheDocument()` at all if the `getByBlah` explodes anyway?  Why not just write `getByBlah(...)` in isolation if it's going to explode anyway?  It's not like the `expect` it's getting passed to will be able to do anything if it throws an error.  I may well be missing an obvious thing here!

Reply:
Yep you're right there - the `getBy` variants are better for debugging purposes as they dump out the HTML.

Kent Dodds mentions it here: https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#using-get-variants-as-assertions

Main benefit is just to communicate intent - i.e. that we're asserting something. I'd agree with him on this as just having free-floating `getBy` selectors feels a bit weird from a readability perspective.